### PR TITLE
Fix code analysis errors

### PR DESCRIPTION
**Pull request by JamesNK**
* Add static when possible
* Remove unused private members
* Remove unused out parameters

### CODE REVIEW

**Comment by JamesNK** [comment]
I don't know who owns all these areas so I added GH suggested reviewers 😋

File: src/Aspire.Hosting/Dcp/ApplicationExecutor.cs [hunk 1]
```diff
@@ -106,13 +106,6 @@ public async Task StopApplicationAsync(CancellationToken cancellationToken = def
         }
     }
 
-    private async Task CreateContainerSingletonsAsync(CancellationToken cancellationToken = default)
-    {
-        // Find containers that consume no Services--their associated Services can be started in Proxyless mode
-        var toCreate = _appResources.Where(r => r.DcpResource is Container && !r.ServicesConsumed.Any());
-        await CreateContainersAsync(toCreate, cancellationToken).ConfigureAwait(false);
-    }
-
```
Comment (drewnoakes) [inline src/Aspire.Hosting/Dcp/ApplicationExecutor.cs:115]: The calling code was commented out, so ideally we'd either comment this, or delete the comment. Ditto below.

**Review by drewnoakes — APPROVED** [review]
Thanks for filing this. I was just about to.

**Comment by drewnoakes** [comment]
My understanding is they're IDE* diagnostics that only show in VS. I don't know why that's the case. They only appeared for me after updating VS.

**Pull request merged**